src/diffusers/pipelines/marigold/pipeline_marigold_depth.py

```diff
@@ -0,0 +1,1246 @@
+        arg_is_np = isinstance(image, np.ndarray)
+        if arg_is_np:
+            image = torch.tensor(image)
+        if image.dtype == torch.uint8:
+            image = image.float() / 255
+        else:
+            image = image.float()
+
+        if cmap != "Spectral":
+            raise ValueError("Only 'Spectral' color map is available without installing matplotlib.")
+
+        _Spectral_data = (  # Copied from matplotlib/_cm.py
+            (0.61960784313725492, 0.003921568627450980, 0.25882352941176473),  # 0.0 -> [0]
+            (0.83529411764705885, 0.24313725490196078, 0.30980392156862746),
+            (0.95686274509803926, 0.42745098039215684, 0.2627450980392157),
+            (0.99215686274509807, 0.68235294117647061, 0.38039215686274508),
+            (0.99607843137254903, 0.8784313725490196, 0.54509803921568623),
+            (1.0, 1.0, 0.74901960784313726),
+            (0.90196078431372551, 0.96078431372549022, 0.59607843137254901),
+            (0.6705882352941176, 0.8666666666666667, 0.64313725490196083),
+            (0.4, 0.76078431372549016, 0.6470588235294118),
+            (0.19607843137254902, 0.53333333333333333, 0.74117647058823533),
+            (0.36862745098039218, 0.30980392156862746, 0.63529411764705879),  # 1.0 -> [K-1]
+        )
+
+        cmap = torch.tensor(_Spectral_data, dtype=torch.float, device=image.device)  # [K,3]
+        K = cmap.shape[0]
+
+        pos = image.clamp(min=0, max=1) * (K - 1)
+        left = pos.long()
+        right = (left + 1).clamp(max=K - 1)
+
+        d = (pos - left.float()).unsqueeze(-1)
+        left_colors = cmap[left]
+        right_colors = cmap[right]
+
+        out = (1 - d) * left_colors + d * right_colors
+
+        if bytes:
+            out = (out * 255).to(torch.uint8)
+
+        if arg_is_np:
+            out = out.numpy()
+
+        return out
+
+    if force_method is None and torch.is_tensor(image) and cmap == "Spectral":
+        return method_custom(image, cmap, bytes)
+
+    out = None
+    if force_method != "custom":
+        out = method_matplotlib(image, cmap, bytes)
+
+    if force_method == "matplotlib" and out is None:
+        raise ImportError("matplotlib is not installed")
+
+    if out is None:
+        out = method_custom(image, cmap, bytes)
+
+    return out
+
+
+def visualize_depth(
+    depth: torch.FloatTensor,
+    color_map: str = "Spectral",
+    vis_min: float = 0.0,
+    vis_max: float = 1.0,
+):
+    assert depth.dim() == 4 and depth.shape[:2] == (1, 1)
+
+    if vis_max <= vis_min:
+        raise ValueError(f"Invalid depth range for colorization: [{vis_min}, {vis_max}]")
+
+    if vis_min != 0.0 or vis_max != 1.0:
+        depth = (depth - vis_min) / (vis_max - vis_min)
+    depth = depth.squeeze(1).squeeze(0)  # [H,W]
+
+    visualization = colormap(depth, cmap=color_map, bytes=True)  # [H,W,3]
+    visualization = Image.fromarray(visualization.cpu().numpy())
+
+    return visualization
```

Comment:
This should likely go to `export_utils.py`:
https://github.com/huggingface/diffusers/blob/main/src/diffusers/utils/export_utils.py

Reply:
Fixed